src/main/java/com/insightfullogic/honest_profiler/core/sources/VirtualMachine.java

```diff
@@ -63,6 +66,19 @@ public boolean isAgentLoaded()
         return agentLoaded;
     }
 
+    public LogSource getLogSourceFromVmArgs()
+    {
+        int agentPathStart = vmArgs.indexOf("-agentpath") + "-agentpath".length();
+        int agentPathEnd = vmArgs.indexOf(' ', agentPathStart);
+        String agentPath = vmArgs.substring(agentPathStart, agentPathEnd);
+        int logPathStart = agentPath.indexOf("logPath=") + "logPath=".length();
+        int commaPos = agentPath.indexOf(",", logPathStart);
+        int spacePos = agentPath.indexOf(' ', logPathStart);
+        int logPathEnd = commaPos > 0 ? commaPos : (spacePos > 0 ? spacePos : agentPath.length());
+
+        return new FileLogSource(new File(agentPath.substring(logPathStart, logPathEnd)));
+    }
```

Comment:
Fixed in upcoming commit. I hardened the code, by introducing early checks if what we're looking for is present at all, and revert to original getLogSource() if it isn't. Also, the final File is checked for existence, and if it doesn't, revert to the original getLogSource() as well.